### PR TITLE
[BEAM-2338] Fix the limit counter in gcsio reads

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @chamikaramj PTAL

### CODE REVIEW

File: sdks/python/apache_beam/io/gcp/gcsio.py [hunk 1]
```diff
@@ -392,7 +392,7 @@ def size_of_files_in_glob(self, pattern, limit=None):
         if fnmatch.fnmatch(item.name, name_pattern):
           file_name = 'gs://%s/%s' % (item.bucket, item.name)
           file_sizes[file_name] = item.size
-        counter += 1
+          counter += 1
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/gcsio.py]: Can we add a test ?

Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/gcsio.py]: The tests just mock GCS so don't really catch anything here.

Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/gcsio.py]: Done.

**Comment by sb2nov** [comment]
R: @chamikaramj PTAL again

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Comment by chamikaramj** [comment]
Will merge after tests pass.

**Comment by sb2nov** [comment]
Tests are green, should be okay to merge now.

Thanks

**Pull request closed without merging**